sdk/cmake/msvc.cmake

```diff
@@ -409,21 +409,15 @@ endmacro()
 # PSEH workaround
 set(PSEH_LIB "pseh")
 
-# Use a full path for the x86 version of ml when using x64 VS.
-# It's not a problem when using the DDK/WDK because, in x64 mode,
-# both the x86 and x64 versions of ml are available.
-if(ARCH STREQUAL "amd64")
+# Setup MASM for compiling 16-bit x86 ASM code on non-x86 ports.
+if(NOT ARCH STREQUAL "i386")
     if((MSVC_VERSION LESS_EQUAL 1900) AND (DEFINED ENV{VCINSTALLDIR}))
         set(CMAKE_ASM16_COMPILER $ENV{VCINSTALLDIR}/bin/ml.exe)
     elseif(DEFINED ENV{VCToolsInstallDir})
-        set(CMAKE_ASM16_COMPILER $ENV{VCToolsInstallDir}/bin/HostX86/x86/ml.exe)
+        set(CMAKE_ASM16_COMPILER $ENV{VCToolsInstallDir}/bin/Host$ENV{VSCMD_ARG_HOST_ARCH}/x86/ml.exe)
```

Comment:
Is this `VSCMD_ARG_HOST_ARCH` correctly defined for all supported MSVC compilers (in particular, maybe 2013? and 2017; it should certainly be there in 2019) ?

Reply:
Supported by and only used for VS 2017+, older versions use `VCINSTALLDIR` path.